resources/cron.rb

```diff
@@ -0,0 +1,72 @@
+#
+# Cookbook:: chef-client
+# resource:: chef_client_cron
+#
+# Copyright:: 2020, Chef Software, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+resource_name :chef_client_cron
+
+property :user, String, default: 'root'
+
+property :minute, [String, Integer], default: 0
+property :hour, [String, Integer], default: '0,4,8,12,16,20'
+property :weekday, [String, Integer], default: '*'
+property :mailto, String
+
+property :job_name, String, default: 'chef-client'
+property :splay, [Integer, String], default: 300
+
+property :env_vars, Hash
+
+property :config_directory, String, default: '/etc/chef'
+property :log_directory, String, default: '/var/log/chef'
+property :log_file_name, String, default: 'client.log'
+property :append_log_file, [true, false], default: false
+property :chef_binary_path, String, default: '/opt/chef/bin/chef-client'
+property :daemon_options, Array, default: []
+
+action :add do
+  cron_d new_resource.job_name do
+    minute  new_resource.minute
+    hour    new_resource.hour
+    weekday new_resource.weekday
+    mailto  new_resource.mailto if new_resource.mailto
+    user    new_resource.user
+    command cron_command
+  end
+end
+
+action :remove do
+  cron_d new_resource.job_name do
+    action :delete
+  end
+
+  cron new_resource.job_name do
+    action :delete
+  end
```

Comment:
does that actually map properly?

Reply:
It should nuke a legacy named chef-client job if it's there